File: DownloadFiles/TGNet/Chapter03/Chapter3DemoCode/C#/TextReadUtility/TextReadUtility/frmTextRead.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using System.IO;

#endregion

namespace TextReadUtility
{
    partial class frmTextRead : Form
    {
        public frmTextRead()
        {
            InitializeComponent();
        }

        private void btnRead_Click(object sender, EventArgs e)
        {
            string strFile;
            int MyPos = 1;
            Int16 iCount = 0;
            StreamReader reader = File.OpenText(txtFile.Text);
            strFile = reader.ReadToEnd();
            do
            {
                MyPos = strFile.IndexOf(txtFindString.Text, MyPos + 1);
                if (MyPos > 0)
                {
                    iCount += 1;
                }
            } while (!(MyPos == -1));
            lblResult.Text = "Number of times found: iCount = " + iCount;

        }
    }
}

File: DownloadFiles/TGNet/Chapter03/Chapter3DemoCode/VB/TextReadUtility/TextReadUtility/frmTextRead.vb
Imports System.IO

Public Class frmTextRead

    Private Sub btnRead_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnRead.Click
        Dim strFile As String
        Dim MyPos As Integer = 1
        Dim iCount As Int16 = 0

        ' Open the stream and read it back.
        Dim reader As StreamReader = File.OpenText(txtFile.Text)
        strFile = reader.ReadToEnd()

        Do
            MyPos = strFile.IndexOf(txtFindString.Text, MyPos + 1)
            If MyPos > 0 Then iCount += 1
        Loop Until (MyPos = -1)

        lblResult.Text = "Number of times found: iCount = " & iCount


    End Sub

End Class
